mono/metadata/appdomain.c

```diff
@@ -1598,9 +1598,39 @@ mono_domain_fire_assembly_load (MonoAssemblyLoadContext *alc, MonoAssembly *asse
 	void *params [1];
 	params [0] = MONO_HANDLE_RAW (reflection_assembly);
 	mono_runtime_invoke_handle_void (assembly_load_method, appdomain, params, error);
+#endif
+
+exit:
+	HANDLE_FUNCTION_RETURN ();
+}
+
+static void
+mono_domain_fire_assembly_load (MonoAssemblyLoadContext *alc, MonoAssembly *assembly, gpointer user_data, MonoError *error_out)
+{
+	ERROR_DECL (error);
+	MonoDomain *domain = mono_alc_domain (alc);
+
+	if (!MONO_BOOL (domain->domain))
+		goto leave; // This can happen during startup
+
+	if (mono_runtime_get_no_exec ())
+		goto leave;
+
+#ifdef ASSEMBLY_LOAD_DEBUG
+	fprintf (stderr, "Loading %s into domain %s\n", assembly->aname.name, domain->friendly_name);
+#endif
```

Comment:
nit: if you're going to have one more go around on this PR, change this to a `mono_trace` and also print the alc ptr - might come in useful.

Reply:
I noticed this after I finished the CI cycle and marked it to do today. I dunno if that define is used elsewhere in the source but it should probably be excised completely.

Reply:
Looking at it now, I'll just change it here and put up with another CI cycle.